### PR TITLE
drivers: usb: udc: add numaker m2l31x usbd controller

### PR DESCRIPTION
**Pull request by ccli8**
This is to support Nuvoton NuMaker M2L31 series SoC usbd controller.

Verified samples include:

- zephyr/samples/subsys/usb/cdc_acm
- zephyr/samples/subsys/usb/console
- zephyr/samples/subsys/usb/mass

### CODE REVIEW

**Review by fabiobaltieri — APPROVED** [review]

**Review by cyliangtw — APPROVED** [review]

**Comment by ccli8** [comment]
Any further comment?

**Comment by fabiobaltieri** [comment]
Sorry @ccli8 this was misassigned, should have gone to Numicro maintainers, feel free to point it out if it happens again, PRs are not going to show up in the release team merge dashboard until approved.

**Pull request merged**
